### PR TITLE
docs/esp32/quickref: Add info about LAN

### PR DESCRIPTION
**Pull request by spacemanspiff2007**
docs/esp32/quickref: Add info about LAN

This is a small docs extension which shows some information about 
LAN and pinouts for popular boards.

Signed-off-by: Sebastian Gerber <github@sgerber.de>

### CODE REVIEW

**Comment by spacemanspiff2007** [comment]
Can someone run the checks again? I edited the commit msg.

**Comment by spacemanspiff2007** [comment]
@dpgeorge Do I still have to do something or is this good to get merged?

File: docs/esp32/quickref.rst [hunk 1]
```diff
@@ -102,6 +102,32 @@ Once the network is established the :mod:`socket <usocket>` module can be used
 to create and use TCP/UDP sockets as usual, and the ``urequests`` module for
 convenient HTTP requests.
 
+LAN
+^^^
```
Comment (dpgeorge) [inline docs/esp32/quickref.rst:106]: this should match other headings at this level, ie use `===`

File: docs/esp32/quickref.rst [hunk 2]
```diff
@@ -102,6 +102,32 @@ Once the network is established the :mod:`socket <usocket>` module can be used
...
+
+    import network
+
+    lan = network.LAN(mdc=PIN_MDC, ...)   # Set the pin configuration
+    lan.active(True)                      # activate the interface
+    lan.ifconfig()                        # get the interface's IP/netmask/gw/DNS addresses
+
+
+Pinouts for popular boards
+^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (dpgeorge) [inline docs/esp32/quickref.rst:118]: this heading is not needed, it's enough to just make this section a paragraph (to keep the quickref short and to the point)

File: docs/esp32/quickref.rst (hunk 1 above)
Comment (dpgeorge) [reply docs/esp32/quickref.rst:106]: aah, did you intend for this to be a subheading under `Networking`?

Comment (spacemanspiff2007) [reply docs/esp32/quickref.rst:106]: yes - exactly. The idea was that unter networking there is WLAN and LAN so the user can jump directly to the section

**Review by spacemanspiff2007 — COMMENTED** [review]
Like this?

File: docs/esp32/quickref.rst (hunk 1 above)
Comment (dpgeorge) [reply docs/esp32/quickref.rst:117]: ok

File: docs/esp32/quickref.rst [hunk 3]
```diff
@@ -110,6 +113,29 @@ calling ``wlan.config(reconnects=n)``, where n are the number of desired reconne
...
+    lan = network.LAN(mdc=PIN_MDC, ...)   # Set the pin configuration
+    lan.active(True)                      # activate the interface
+    lan.ifconfig()                        # get the interface's IP/netmask/gw/DNS addresses
+
+
+These are working configurations for LAN interfaces of popular boards:
+
+Olimex ESP32-GATEWAY::
+
+   network.LAN(mdc=machine.Pin(23), mdio=machine.Pin(18), power=machine.Pin(5), phy_type=network.PHY_LAN8720, phy_addr=0, clock_mode=network.ETH_CLOCK_GPIO17_OUT)
```
Comment (dpgeorge) [inline docs/esp32/quickref.rst:132]: note that `clock_mode` was removed for IDF 4.x LAN, so these configurations will not work anymore

Comment (spacemanspiff2007) [reply docs/esp32/quickref.rst]: Ethernet didn't work yet for IDF4 when I tried the configurations.

Comment (dpgeorge) [reply docs/esp32/quickref.rst]: did you get a chance to try the new code out now with these configuration lines?

**Comment by spacemanspiff2007** [comment]
Do I still have to do something?

**Comment by dpgeorge** [comment]
> Do I still have to do something?

Did you try out the configuration code snippets for the Olimex boards with IDF 4.1+?

**Comment by spacemanspiff2007** [comment]
I am getting the following error:
```
Loaded configuration
E (3575) emac_esp32: emac_esp32_init(308): reset timeout
E (3575) esp_eth: esp_eth_driver_install(198): init mac failed
Task exception wasn't retrieved
future: <Task> coro= <generator object 'lan_ok' at 3ffefe80>
Traceback (most recent call last):
  File "uasyncio/core.py", line 1, in run_until_complete
  File "connection_wifi.py", line 49, in lan_ok
OSError: esp_eth_driver_install failed
Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.
```
in this line
```python
    lan = network.LAN(mdc=machine.Pin(23), mdio=machine.Pin(18), power=machine.Pin(5), phy_type=network.PHY_LAN8720, phy_addr=0)
```

Am I doing something wrong because it seems that LAN support is not available with 1.16.

**Comment by luke-jr** [comment]
This seems to get connected with [Wireless-Tag's WT32-ETH01](http://www.wireless-tag.com/portfolio/wt32-eth01/) ([also available from Seeed Studio](https://www.seeedstudio.com/Ethernet-module-based-on-ESP32-series-WT32-ETH01-p-4736.html))

```
import network, machine
machine.Pin(16, machine.Pin.OUT).on()
lan = network.LAN(mdc=machine.Pin(23), mdio=machine.Pin(18), phy_type=network.PHY_LAN8720, phy_addr=1, power=None)
lan.active(True)
```

**Comment by jimmo** [comment]
FYI this works for the Espressif Ethernet-Kit_A_V1.2

`lan = network.LAN(id=0, mdc=Pin(23), mdio=Pin(18), power=Pin(5), phy_type=network.PHY_IP101, phy_addr=1)`

@dpgeorge I think we should merge this -- it was not at all obvious to me how to configure the LAN, even reading the source code, so these examples are very useful.

Perhaps just simplify the example down to a single code block:

```py
# Olimex ESP32-GATEWAY
lan = network.LAN(mdc=machine.Pin(23), mdio=machine.Pin(18), power=machine.Pin(5), phy_type=network.PHY_LAN8720, phy_addr=0)
# Olimex ESP32-POE::
lan = network.LAN(mdc=machine.Pin(23), mdio=machine.Pin(18), power=machine.Pin(12), phy_type=network.PHY_LAN8720, phy_addr=0)
# Wireless-Tag's WT32-ETH01
lan = network.LAN(mdc=machine.Pin(23), mdio=machine.Pin(18), phy_type=network.PHY_LAN8720, phy_addr=1, power=None)
# Espressif ESP32-Ethernet-Kit_A_V1.2
lan = network.LAN(id=0, mdc=Pin(23), mdio=Pin(18), power=Pin(5), phy_type=network.PHY_IP101, phy_addr=1)
```

**Comment by robert-hh** [comment]
Re-used by PR #10329

**Comment by dpgeorge** [comment]
An updated and extended version of this PR was merged in 30bac47b1256154b8fd10508404068daeda19e71

**Pull request closed without merging**